docs/source/contrib.rst

```diff
@@ -17,17 +17,56 @@ If you like the project and think you could help with making it better, there ar
 Any contribution would be of great help and I will highly appreciate it! If you have any questions, please create a new issue.
 
 
-Pre-commit checks
------------------
+Development process
+-------------------
 
-Make sure all the tests pass and the code pass the coding standard checks::
+Create a fork of the git repository and checkout a new branch from ``master`` branch. 
+The branch name may start with an associated issue number so that we can easily 
+cross-reference them. For example, use ``1234-some-brach-name`` as the name of the branch working to fix issue ``1234``.
+Once you're ready to run a full batterie of tests to your changes, open a pull request.
 
-    tox -p all
+Don't forget to sync your fork once in while to work from the latest revision.
 
-That should be the minimum check to run on your local system.
+Pre-commit checks
+-----------------
+
+Make sure all the unit tests pass and the code pass the coding standard checks.
+
+We use `tox <https://tox.wiki/en/stable/>`_ for running our checks, but you can roughly do the same thing from your python environment. 
+
+.. list-table:: Pre-commit checks
+   :widths: 10 45 45
+   :header-rows: 1
+   
+   * - \
+     - Using `tox`
+     - Using your environment
+   * - Run unit tests
+     - ``tox -e test``
+     - ``pip install '.[test]' && pytest pydoctor``
+   * - Run pyflakes
+     - ``tox -e pyflakes``
```

Comment:
Here you run a linter. You #757 you told me that you don't.

#757 is not fixed. IMHO it would be a good idea to mention your code style even if you do not care about code style. Also the latter is a "tell".

What I do not understand is why pyflakes do not throw any errors. I couldn't find a pyflakes config somewhere. But the code do not follow PEP8 and should throw 100 of errors.

Reply:
This is only about line 47-48.

Reply:
This is a nice explanation and could fit well into a contrib file.